src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBExtensions.cs

```diff
@@ -224,9 +175,151 @@ public static IResourceBuilder<AzureCosmosDBEmulatorResource> WithPartitionCount
+            Locations =
+                {
+                    new CosmosDBAccountLocation
+                    {
+                        LocationName = new IdentifierExpression("location"),
+                        FailoverPriority = 0
+                    }
+                },
+            Tags = { { "aspire-resource-name", infrastructure.AspireResource.Name } }
+        };
+        infrastructure.Add(cosmosAccount);
+
+        foreach (var database in azureResource.Databases)
+        {
+            var cosmosSqlDatabase = new CosmosDBSqlDatabase(Infrastructure.NormalizeBicepIdentifier(database.Name))
+            {
+                Parent = cosmosAccount,
+                Name = database.Name,
+                Resource = new CosmosDBSqlDatabaseResourceInfo()
+                {
+                    DatabaseName = database.Name
+                }
+            };
+            infrastructure.Add(cosmosSqlDatabase);
+
+            foreach (var container in database.Containers)
+            {
+                var cosmosContainer = new CosmosDBSqlContainer(Infrastructure.NormalizeBicepIdentifier(container.Name))
+                {
+                    Parent = cosmosSqlDatabase,
+                    Name = container.Name,
+                    Resource = new CosmosDBSqlContainerResourceInfo()
+                    {
+                        ContainerName = container.Name,
+                        PartitionKey = new CosmosDBContainerPartitionKey { Paths = [container.PartitionKeyPath] }
+                    }
+                };
+                infrastructure.Add(cosmosContainer);
+            }
+        }
+
+        if (azureResource.UseAccessKeyAuthentication)
+        {
+            cosmosAccount.DisableLocalAuth = false;
+
+            var kvNameParam = new ProvisioningParameter(AzureBicepResource.KnownParameters.KeyVaultName, typeof(string));
+            infrastructure.Add(kvNameParam);
+
+            var keyVault = KeyVaultService.FromExisting("keyVault");
+            keyVault.Name = kvNameParam;
+            infrastructure.Add(keyVault);
+
+            var secret = new KeyVaultSecret("connectionString")
+            {
+                Parent = keyVault,
+                Name = "connectionString",
+                Properties = new SecretProperties
+                {
+                    Value = BicepFunction.Interpolate($"AccountEndpoint={cosmosAccount.DocumentEndpoint};AccountKey={cosmosAccount.GetKeys().PrimaryMasterKey}")
+                }
+            };
+            infrastructure.Add(secret);
+        }
+        else
+        {
+            cosmosAccount.DisableLocalAuth = true;
+
+            var principalIdParameter = new ProvisioningParameter(AzureBicepResource.KnownParameters.PrincipalId, typeof(string));
+            infrastructure.Add(principalIdParameter);
+
+            cosmosAccount.CreateRoleAssignment(CosmosDBBuiltInRole.DocumentDBAccountContributor, RoleManagementPrincipalType.ServicePrincipal, principalIdParameter);
```

Comment:
Why isn't this generating a prinicpalType like all the other Azure integrations?

https://github.com/dotnet/aspire/blob/882ae2be3e02edbfc1d222ee97fe2da4eea3fe99/src/Aspire.Hosting.Azure.AppConfiguration/AzureAppConfigurationExtensions.cs#L38-L43

Reply:
The method to create the role on cosmos doesn't take any. It's working though, you think it must still be missing?

Reply:
Yes this should be a parameter, @tg-msft?

Reply:
I must have been on drugs, it's right there now. Weird as I remember changing this code as it was not working ...